contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -975,29 +979,43 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
 
     PaymentReceipt memory receipt = _calculatePaymentAmount(hotVars, paymentParams);
 
-    uint96 balance = upkeep.balance;
-    uint96 payment = receipt.gasCharge + receipt.premium;
+    // this balance uses the token's decimal places
+    uint96 balanceXDecimals = upkeep.balance;
+    // adjust to 18 decimal places to be comparable with the following payment
+    uint96 balance18Decimals = SafeCast.toUint96(
+      balanceXDecimals * 10 ** (18 - paymentParams.billingTokenParams.decimals)
```

Comment:
this will underflow and revert if the billing token uses >18 decimals (no idea how common this is, but I think we might as well support it!)